src/Bowling.js

```diff
@@ -0,0 +1,144 @@
+
+"use strict";
+
+const ALL_PINS = 10;
+
+function Bowling() {
+  this.score = 0;
+  this.frameKey;
+  this.frameScore;
+  this.spare = false;
+  this.strike = false;
+  this.input 
+  //this.rollCount = 1   maybe this should be instead of passing rollCount;
+}
+
+function getRndInteger(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+Bowling.prototype.runCardMaking = function (input) {
+  var rollCount;
+  var frameCount;
+  var pins;
+  var score;
+  this.makeCardTemplate();
+  for (frameCount = 1; frameCount < 11; frameCount++) {
+    this.frameKey = frameCount;
+    for (rollCount = 1; rollCount < 3; rollCount++) {
+      if (this.isSkipped(rollCount)) {
+        pins = "x";
+        score = this.calculateScore(0);
+      } else {
+       
+        pins = this.input;
+
+        this.switchStrike(rollCount, pins)
+         
+        this.scoreStrike(rollCount, pins);
+        score = this.calculateScore(pins);
+      }
+      this.fillCard(pins, score, rollCount);
+
+      if (this.strike === false && this.spare === true) {
```

Comment:
Throughout this class, you rely a lot on state (or instance variables). This makes the understanding more difficult.
What you have now is something like:
- Method A calculates something, and stores it in X
- Later in the function, you read the value at X.

This is not very good design, because you have to always make sure to update all values before you read from them. Instead, you should be doing:
- call method A that calculates a value, and returns it. 

More generally, instance variables should be kept to a minimum, and to things that describe the current state of the object - not the current state of your computation! Local variable are here for that.

Reply:
This is actually a thing I see a lot with Makers devs.